### PR TITLE
Added documentation for new rule CA1847 - Use string.Contains(char) w…

### PR DESCRIPTION
**Pull request by MeikTranel**
## Summary

Adds required documentation for rule CA1847

Required documentation for https://github.com/dotnet/runtime/issues/47180 (PR: https://github.com/dotnet/roslyn-analyzers/pull/4908/files)

## Related
- https://github.com/dotnet/docs/pull/23101 (rule id changed and i messed by referencing the ID in the branch name - lesson learned 🤣)

### CODE REVIEW

**Comment by MeikTranel** [comment]
@gewarren probably should also get the "Hold for related PR" tag

**Comment by adegeo** [comment]
Changed to draft to prevent any accidental merge 😁

**Comment by gewarren** [comment]
Has the rule ID changed again? It looks like CA1846 was used for [prefer AsSpan](https://github.com/dotnet/roslyn-analyzers/pull/4806).

**Comment by MeikTranel** [comment]
Yes, I've just picked up the original PR for requested changes again. I'll update this PR when the roslyn-analyzer PR gets merged.

**Comment by MeikTranel** [comment]
Upstream PR has been merged now. Will be pushing the final changes to the PR later today.

**Comment by MeikTranel** [comment]
@gewarren can we get this PR forward? The roslyn-analyzer PR has been merged for a while now ;)

File: docs/fundamentals/code-analysis/quality-rules/ca1847.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+```csharp
+public bool ContainsLetterI()
+{
+    var testString = "I am a test string.";
+    return testString.Contains("I");
+}
+```
+
+```vb
+Public Function ContainsLetterI as Boolean
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca1847.md:45]:
```suggestion
Public Function ContainsLetterI() As Boolean
```

File: docs/fundamentals/code-analysis/quality-rules/ca1847.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+```csharp
+public bool ContainsLetterI()
+{
+    var testString = "I am a test string.";
+    return testString.Contains('I');
+}
+```
+
+```vb
+Public Function ContainsLetterI as Boolean
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca1847.md:62]:
```suggestion
Public Function ContainsLetterI() As Boolean
```

**Review by adegeo — APPROVED** [review]
LGTM @gewarren

**Pull request merged**